### PR TITLE
Fix typos noticed in conversion utilities

### PR DESCRIPTION
**Pull request by zoddicus**
<hr>

**Author checklist for test code/plans:**

- [x] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [x] New helpers, if any, are documented using `/** doc comments */` and can be found via `helper_index.txt`.
- [x] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)

**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [ ] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [ ] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [ ] Existing (or new) test helpers are used where they would reduce complexity.
- [ ] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

File: src/webgpu/util/conversion.ts [hunk 1]
```diff
@@ -382,7 +382,8 @@ export class Scalar {
 
 /** Create an f32 from a numeric value, a JS `number`. */
 export function f32(value: number): Scalar {
-  return new Scalar(TypeF32, value, new Float32Array([value]));
+  const arr = new Float32Array([value]);
```
Comment (ben-clayton) [inline src/webgpu/util/conversion.ts:385]: We'll want to do this for `i32`, `u32`, `bool` too.

Comment (zoddicus) [reply src/webgpu/util/conversion.ts:385]: Done

File: src/webgpu/util/conversion.ts [hunk 2]
```diff
@@ -458,7 +465,8 @@ export function bool(value: boolean): Scalar {
   // WGSL does not support using 'bool' types directly in storage / uniform
   // buffers, so instead we pack booleans in a u32, where 'false' is zero and
   // 'true' is any non-zero value.
-  return new Scalar(TypeBool, value, new Uint32Array([value ? 1 : 0]));
+  const arr = new Uint32Array([value ? 1 : 0]);
+  return new Scalar(TypeBool, arr[0], arr);
```
Comment (ben-clayton) [inline src/webgpu/util/conversion.ts:469]: Sorry, this one really should use `value`, as this is no longer a boolean.

Comment (zoddicus) [reply src/webgpu/util/conversion.ts]: Done

**Review by ben-clayton — APPROVED** [review]
Thank you!

File: src/webgpu/util/conversion.ts [hunk 3]
```diff
@@ -391,21 +392,24 @@ export function f32Bits(bits: number): Scalar {
 }
 /** Create an f16 from a bit representation, a uint16 represented as a JS `number`. */
 export function f16Bits(bits: number): Scalar {
-  const arr = new Uint32Array([bits]);
+  const arr = new Uint16Array([bits]);
   return new Scalar(TypeF16, float16BitsToFloat32(bits), arr);
```
Comment (dneto0) [inline src/webgpu/util/conversion.ts:396]: Should the arg to float16BitsToFloat32 be arr[0] instead of bits?

Comment (dneto0) [inline src/webgpu/util/conversion.ts:396]: Should the arg to float16BitsToFloat32 be arr[0] instead of bits?

Comment (dneto0) [reply src/webgpu/util/conversion.ts:396]: oh,  float16BitsToFloat32 only pays attention to the bottom 16 bits of its argument.  never mind.  :-)

**Pull request merged**